_studio/mfx_lib/decode/vp9/src/mfx_vp9_dec_decode_hw.cpp

```diff
@@ -749,14 +746,12 @@ mfxStatus VideoDECODEVP9_HW::GetVideoParam(mfxVideoParam *par)
     return MFX_ERR_NONE;
 }
 
-void VideoDECODEVP9_HW::UpdateVideoParam(mfxVideoParam *par, VP9DecoderFrame const & frameInfo)
+mfxStatus VideoDECODEVP9_HW::UpdateVideoParam(mfxVideoParam *par, VP9DecoderFrame const & frameInfo)
```

Comment:
why can't you pass ```mfxVideoParam``` by reference? This will allow you to get rid of pointer check and error status return.

Or if you still prefer to return error status, you need to add status check after ```UpdateVideoParam``` call